### DISCUSSION TITLE
State of LSP and Vala

### DISCUSSION THREAD

**Discussion by chances** [chances/vala-language-server-protocol#10] [General]
Spawned from https://github.com/elementary/code/issues/220

- [SDKs for the LSP](https://microsoft.github.io/language-server-protocol/implementors/sdks/)
- https://github.com/OmniSharp/csharp-language-server-protocol
- [Language Server Protocol Support](https://wiki.gnome.org/Projects/Vala/Tools#Language_Server_Protocol_Support) in Vala - GNOME wiki
- [Language Server Protocol for Vala](https://wiki.gnome.org/Projects/Vala/LanguageServerSpecifications) - GNOME wiki
  - https://github.com/Prince781/vala-language-server

**Comment by Prince781** [comment]
This project looks interesting. Is your goal to make a general GObject library for LSP, to be used by Code and possibly GNOME Builder?

**Comment by chances** [comment]
@Prince781 Exactly! 😄 My first priority is integration with [Code](https://github.com/elementary/code), but I want the library to be generic enough for other use cases as well.

**Comment by chances** [comment]
@Prince781 Do you think, if at all, this project could integrate with https://github.com/Prince781/vala-language-server?

~~I didn't find this your repo during my initial search for extant implementations of the protocol. Yours is not listed on the [GNOME Wiki](https://wiki.gnome.org/Projects/Vala/Tools#Language_Server_Protocol_Support)?~~ (Turns out it _is_ listed as https://github.com/benwaffle/vala-language-server.)

**Comment by Prince781** [comment]
I don't see this project integrating with VLS or any language server. Rather, I see it as providing a general implementation of the LSP that can be used by any editor.

**Comment by Prince781** [comment]
I would be interested in helping you out, though. I've had a [similar idea](https://github.com/Prince781/lsp-glib) but I have other things to work on (namely VLS and other areas of Vala). 

If the idea becomes a library implementation for LSP, I think the project name should be reconsidered.

**Comment by chances** [comment]
@Prince781 My goal with the project is a generic library implementing the LSP protocol, much like the other [LSP SDKs](https://microsoft.github.io/language-server-protocol/implementors/sdks/), such that it may be used in _both_ LSP client and server implementations. I think the library could help reduce the maintenance burden of VLS' [LSP protocol implementation.](https://github.com/Prince781/vala-language-server/blob/master/src/protocol.vala)

I would appreciate your help. :smile:

Regarding the project's name, I based it off of [this](https://github.com/OmniSharp/csharp-language-server-protocol) C# LSP SDK. Why do you think the current name isn't suitable?

**Comment by Prince781** [comment]
Because Vala libraries aren't written just for Vala, but can be used from C and any other language that has bindings (Python, Rust, etc). I was thinking of `lsp-glib` or `langserv-glib` as other GLib-based libraries are similarly named (jsonrpc-glib, json-glib, libgit-glib, telepathy-glib, etc)
